### DISCUSSION TITLE
How to manipulate data before saving or getting by Rest API Directus side?

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#7807] [Archive]
How to manipulate data before saving or getting by Rest API Directus side?

**Comment by rijkvanzanten** [comment]
Hooks! https://docs.directus.io/guides/api-hooks/ 🙂

**Reply by biapar** [reply]
Not with controllers or events overloads?

**Reply by rijkvanzanten** [reply]
Hooks is our terms for "events overloads". Custom "controllers" can be added through custom endpoints: https://docs.directus.io/guides/api-endpoints/
